src/sinks/util/sink.rs

```diff
@@ -706,6 +709,36 @@ where
     }
 }
 
+// === Response ===
+
+pub trait Response: fmt::Debug {
+    fn is_successful(&self) -> bool {
+        true
+    }
+}
+
+impl Response for () {}
+impl<'a> Response for &'a str {}
+
+#[cfg(feature = "rusoto_kinesis")]
+impl Response for rusoto_kinesis::PutRecordsOutput {}
+#[cfg(feature = "rusoto_s3")]
+impl Response for rusoto_s3::PutObjectOutput {}
+#[cfg(feature = "rusoto_firehose")]
+impl Response for rusoto_firehose::PutRecordBatchOutput {}
+
+impl<T: fmt::Debug> Response for http::response::Response<T> {
+    fn is_successful(&self) -> bool {
+        self.status().is_success()
+    }
+}
+
+impl<T: fmt::Debug> Response for http02::response::Response<T> {
+    fn is_successful(&self) -> bool {
+        self.status().is_success()
+    }
+}
```

Comment:
These could go into `src/sinks/util/http.rs` and `src/sinks/util/http2.rs` respectively, though I don't think it's a big deal.